src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -1478,7 +1567,13 @@ internal int UpperBound
         }
 
         private int _current;
-        public object Current
+
+        Object IEnumerator.Current
```

Comment:
Can you explain why this method was added? It seems like we only need the explicit method impl or the other, but not both.

Reply:
I saw it in CoreFX implementations ( [sample](https://github.com/dotnet/corefxlab/blob/b5a998d2758b8ee968dad324f104c9310057eda3/src/System.IO.Pipelines.Text.Primitives/SplitEnumerator.cs#L30) ). I don't remember exactly - it seems I'm checking that it doesn't work at all without this method if we use enheritance and call by IEnumerator interface.

Reply:
@lzybkr What is your conclusion?

Reply:
The example you point to is needed to implement the strongly typed interface `IEnumerable<T>`. If you wanted to follow that pattern, that's fine, but you would replace the `CurrentValue` property with the `Current` property.

Reply:
Fixed.